pkg/asset/agent/gencrypto/authconfig.go

```diff
@@ -111,20 +160,161 @@ func keyPairPEM() (string, string, error) {
 	return pubKeyPEM.String(), privKeyPEM.String(), nil
 }
 
-func localJWTForKey(id string, privateKkeyPem string) (string, error) {
+// generateToken returns a JWT token based on the private key.
+func generateToken(token *jwt.Token, privateKkeyPem string) (string, error) {
 	priv, err := jwt.ParseECPrivateKeyFromPEM([]byte(privateKkeyPem))
 	if err != nil {
 		return "", err
 	}
+	tokenString, err := token.SignedString(priv)
+	if err != nil {
+		return "", err
+	}
+	return tokenString, nil
+}
+
+func initClient(kubeconfig string) (*kubernetes.Clientset, error) {
+	var err error
+	var config *rest.Config
+	if kubeconfig != "" {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	} else {
+		config, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		return nil, err
+	}
 
-	token := jwt.NewWithClaims(jwt.SigningMethodES256, jwt.MapClaims{
-		string(InfraEnvKey): id,
-	})
+	k8sclientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
 
-	tokenString, err := token.SignedString(priv)
+	return k8sclientset, err
+}
+
+func (a *AuthConfig) createOrUpdateAuthTokenSecret(kubeconfigPath string) error {
+	k8sclientset, err := initClient(kubeconfigPath)
+	if err != nil {
+		return err
+	}
+	// check if secret exists
+	retrievedSecret, err := k8sclientset.CoreV1().Secrets(authTokenSecretNamespace).Get(context.Background(), authTokenSecretName, metav1.GetOptions{})
+	// if the secret does not exist
+	if err != nil {
+		if errors.IsNotFound(err) {
+			err = a.createSecret(k8sclientset)
+			if err != nil {
+				return err
+			}
+			return nil
+		}
+		// Other errors while trying to get the secret
+		return fmt.Errorf("unable to retrieve secret %s/%s: %w", authTokenSecretNamespace, authTokenSecretName, err)
+	}
+
+	// if the secret exists in the cluster, get the token
+	retrievedToken, err := extractAuthTokenFromSecret(retrievedSecret)
+	if err != nil {
+		return err
+	}
+	expiryTime, err := ParseExpirationFromToken(retrievedToken)
```

Comment:
Not a blocking point for this PR, but I noticed that the golang-jwt offers some interesting methods for [validating](https://github.com/openshift/installer/blob/b80f72c853e4378de3f96f0c3d88e6aa208358c4/vendor/github.com/golang-jwt/jwt/v4/map_claims.go#L124) the token, it looks like they could be used to reduce/simplify the code here